### DISCUSSION TITLE
In Efficient_ad, Dose the lr not need to be updated?

### DISCUSSION THREAD

**Discussion by rachelYY12** [open-edge-platform/anomalib#2079] [Q&A]
def configure_optimizers(self) -> optim.Optimizer:
        optimizer = optim.Adam(
            list(self.model.student.parameters()) + list(self.model.ae.parameters()),
            lr=self.lr,
            weight_decay=self.weight_decay,
        )
        num_steps = min(
            self.trainer.max_steps, self.trainer.max_epochs * len(self.trainer.datamodule.train_dataloader())
        )
        scheduler = optim.lr_scheduler.StepLR(optimizer, step_size=int(0.95 * num_steps), gamma=0.1)
        return {"optimizer": optimizer, "lr_scheduler": scheduler}

I find lr upgrade after one epoch in pytorch_lightning，but  step_size=int(0.95 * num_steps) is larger than max_epochs，Does the lr not need to be updatated？

**Comment by blaz-r** [comment]
It looks like you are using an older version of the anomalib where this was not properly addressed, as the code for EfficientAD should look like this:
https://github.com/openvinotoolkit/anomalib/blob/14ab5662dac4bd272c1d4c543470c842c30dade2/src/anomalib/models/image/efficient_ad/lightning_model.py#L217-L231
